src/components/Message/Message.tsx

```diff
@@ -0,0 +1,266 @@
+import React, { useCallback, useContext } from 'react';
+
+import { MessageSimple } from './MessageSimple';
+import {
+  PinPermissions,
+  useActionHandler,
+  useDeleteHandler,
+  useEditHandler,
+  useFlagHandler,
+  useMentionsHandler,
+  useMuteHandler,
+  useOpenThreadHandler,
+  usePinHandler,
+  useReactionHandler,
+  useRetryHandler,
+  useUserHandler,
+  useUserRole,
+} from './hooks';
+import {
+  areMessagePropsEqual,
+  defaultPinPermissions,
+  getMessageActions,
+  MESSAGE_ACTIONS,
+} from './utils';
+
+import { ChannelContext } from '../../context';
+
+import type {
+  DefaultAttachmentType,
+  DefaultChannelType,
+  DefaultCommandType,
+  DefaultEventType,
+  DefaultMessageType,
+  DefaultReactionType,
+  DefaultUserType,
+  UnknownType,
+} from '../../../types/types';
+import type { MessageProps, TranslationContextValue } from 'types';
```

Comment:
we've stopped using the import `from 'types';` and import them from where the type is actually declared now. we're moving towards not having the `index.d.ts`